### PR TITLE
Enable nullability in ContainerInfo

### PR DESCRIPTION
**Pull request by gpetrou**
## Proposed changes

- Enable nullability in ContainerInfo.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/winforms/pull/6473)

### CODE REVIEW

File: src/System.Windows.Forms/src/System/Windows/Forms/Layout/TableLayout.ContainerInfo.cs [hunk 1]
```diff
@@ -308,7 +306,7 @@ public LayoutInfo[] FixedChildrenInfo
                     Debug.Assert(ChildInfoValid, "Fetched invalid information");
                     // we only get this in a cached scenario - so we don't have to worry about caching it.
                     LayoutInfo[] fixedChildren = new LayoutInfo[_countFixedChildren];
-                    if (HasChildWithAbsolutePositioning)
+                    if (HasChildWithAbsolutePositioning && _childInfo is not null)
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms/src/System/Windows/Forms/Layout/TableLayout.ContainerInfo.cs:309]: I'm not sure that it's valid to have `_childInfo == null` when we know that we have ` _countFixedChildren > 0`.  I would prefer an exception in this case.

**Review by dreddy-work — DISMISSED** [review]

File: src/System.Windows.Forms/src/System/Windows/Forms/Layout/TableLayout.ContainerInfo.cs [hunk 2]
```diff
@@ -311,7 +309,7 @@ public LayoutInfo[] FixedChildrenInfo
                     if (HasChildWithAbsolutePositioning)
                     {
                         int index = 0;
-                        for (int i = 0; i < _childInfo.Length; i++)
+                        for (int i = 0; i < _childInfo!.Length; i++)
```
Comment (RussKie) [inline src/System.Windows.Forms/src/System/Windows/Forms/Layout/TableLayout.ContainerInfo.cs:312]: Please don't do this.

I believe this should address this:
```cs
            [MemberNotNullWhen(true, nameof(_childInfo))]
            public bool HasChildWithAbsolutePositioning
            {
                get { return _countFixedChildren > 0; }
            }
```

**Review by RussKie — CHANGES_REQUESTED** [review]

**Review by RussKie — DISMISSED** [review]

**Comment by gpetrou** [comment]
@RussKie perhaps you need to manually merge this since the PrintPreviewDialogTests cause too many failures?

**Pull request merged**